src/client/openWeatherClient.js

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+
+const { CancelToken } = axios;
+
+
+class OpenWeatherClient {
+
+    constructor() {
+        this.cancelRequest = null;
+        this.CACHE = new Map();
+
+    }
+
+    getWeather(cityId) {
+
+        const cachedForecasts = this.CACHE.get(cityId);
+
+        return new Promise((resolve, reject) => {
+            if (cachedForecasts && !this.isExpiredRequest(cachedForecasts.requestedTime)) {
+                resolve(cachedForecasts.forecasts);
+            } else {
+                try {
+                    if (this.cancelRequest !== null) {
+                        this.cancelRequest();
+                    }
+                    let requestedTime = new Date();
+                    let forecasts = this.getForecast(cityId);
```

Comment:
forecasts will be a promise object if you don't await it. It means that you are saving promise to cache instead of forecasts array